### PR TITLE
feat: improve UI/UX of playground

### LINKED ISSUE #16: Improve the UI/UX of the Checkout Playground - too many input boxes / confusing layout
# Goal

* Simplify the form
* Improve the payment widget

# Design

https://www.figma.com/design/cEkRPhER2ArCnOwNVDmVVA/Request-Network?node-id=272-3834&t=qxdaonedozKW4bKB-4

### PR DESCRIPTION
**Pull request by aimensahnoun**
Resolves #16 

- Improves the UX of the playground by separating the input fields into separate distinct sections while highlighting the required fields.

Screenshots:
CleanShot 2024-11-06 at 21 17 02
CleanShot 2024-11-06 at 21 17 12
CleanShot 2024-11-06 at 21 17 18

## Release Notes

- **New Features**
	- Introduced a new tabbed interface in the Playground component with three tabs: "Customize Widget," "Seller Billing Info," and "Buyer Billing Info."
	- Added new input fields for seller and buyer information, enhancing user interaction.
	- Included a preview section for the PaymentWidget that dynamically updates based on user input.
	- Added a structured SectionHeader component for improved UI consistency.

- **Bug Fixes**
	- Retained error handling for form validation, ensuring accurate feedback for users.

- **Documentation**
	- Updated internal documentation to reflect the new component structures and functionalities.

### CODE REVIEW

File: src/components/ui/section-header.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+interface SectionHeaderProps {
+  title: string;
+}
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply src/components/ui/section-header.tsx:3]: Out of scope. We can add this later if needed. Resolving as Won't Fix :no_good:

File: src/components/ui/tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    setActiveTab(value);
+    onChange?.(value);
+  };
+
+  return (
+    <TabsContext.Provider value={{ activeTab, setActiveTab: handleTabChange }}>
+      <div className={cn("w-full", className)}>{children}</div>
+    </TabsContext.Provider>
+  );
+};
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply src/components/ui/tabs.tsx:51]: Resolved as Won't Fix :no_good:

File: src/components/ui/tabs.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+const TabSection = ({ value, children }: TabSectionProps) => {
+  const context = React.useContext(TabsContext);
+  if (!context) throw new Error("TabSection must be used within Tabs");
+
+  const { activeTab } = context;
+
+  if (activeTab !== value) return null;
+
+  return <div className="mt-4">{children}</div>;
+};
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply src/components/ui/tabs.tsx:94]: I like the smooth transition idea. 👍 I'm not sure what benefit the aria labels gives.

File: src/components/ui/tabs.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+// Attach components to Tabs
+Tabs.List = TabList;
+Tabs.Section = TabSection;
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply src/components/ui/tabs.tsx:98]: Agreed, but all of this is out of scope for now. Resolving as Won't Fix :no_good:

File: src/components/ui/tabs.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            {tab.label}
+            {activeTab === tab.value && (
+              <div className="absolute bottom-0 left-0 right-0 h-0.5 bg-[#4AC2A1]" />
+            )}
+          </button>
+        ))}
+      </div>
+    </div>
+  );
+};
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply src/components/ui/tabs.tsx:83]: I don't think we need keyboard navigation for now. Resolved as Won't Fix :no_good:

File: src/components/ui/tabs.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (MantisClone) [reply src/components/ui/tabs.tsx:94]: At the risk of sounding insensitive, I don't think we need aria attributes for the playground. Maybe for the payment widget itself. Resolving as Won't Fix :no_good:

File: src/components/Playground.tsx [hunk 6]
```diff
@@ -40,7 +43,7 @@ export const Playground = () => {
       sellerInfo: {},
       buyerInfo: {},
       invoiceNumber: "",
-      enableBuyerInfo: false,
+      enableBuyerInfo: true,
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply src/components/Playground.tsx:46]: I'm not sure the right fix for this but I leave it to @aimensahnoun to decide what to do.

File: src/components/Playground.tsx [hunk 7]
```diff
@@ -144,340 +263,219 @@ const YourComponent = () => {
...
-                      placeholder="Doe"
-                      {...register("sellerInfo.lastName")}
+                    <Label>Product description</Label>
+                    <Textarea
+                      placeholder="Description..."
+                      {...register("productInfo.description")}
                     />
+                    {errors.productInfo?.description?.message && (
+                      <Error>{errors.productInfo.description.message}</Error>
+                    )}
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply src/components/Playground.tsx:296]: Maybe. If we do, we should also have an indicator showing a count of how many characters are left.

I'm willing to say it's not worth the effort right now. Resolving as Won't Fix :no_good:

**Review by MantisClone — APPROVED** [review]
Approved 👍 pending comment resolution 🚧

**Comment by MantisClone** [comment]
@aimensahnoun please don't forget to implement the payment widget UI updates, specified by the design Figma, in a separate PR.

File: src/components/ui/section-header.tsx [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+export function SectionHeader({ title }: SectionHeaderProps) {
+  return (
+    <div className="w-full">
+      <h2 className="text-[#9297A1] uppercase text-sm font-medium tracking-wide mb-4">
+        {title}
+      </h2>
+      <div className="h-px bg-[#D1D7E3] w-full" />
+    </div>
+  );
+}
```
[the remark this answers is not in the document]
Comment (aimensahnoun) [reply src/components/ui/section-header.tsx:14]: This is a custom component but i do not believe these changes are necessary.

File: src/components/Playground.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (aimensahnoun) [reply src/components/Playground.tsx:46]: @coderabbitai earlier implementation had a toggle switch that allows end-user to choose between it being true or not.

We change the behavior to be true by default.

With that said, I've updated the default value in `Zod` to be true by default.

**Pull request merged**